test/files/run/t9814b.scala

```diff
@@ -0,0 +1,54 @@
+trait T {
+  def synchInNestedMethod(): Unit = {
+    def nested = {
+      synchronized {
+        assert(Thread.holdsLock(this))
+      }
+    }
+    nested
+  }
+  val synchInVal: Unit = synchronized {
+    assert(Thread.holdsLock(this))
+  }
+  lazy val synchInLazyVal: Unit = synchronized {
+    assert(Thread.holdsLock(this))
+  }
+  def method() = synchronized {
+    assert(Thread.holdsLock(this))
+  }
+}
+object T extends T
+
+class Foo {
+  def synchInNestedMethod(): Unit = {
+    def nested = {
+      synchronized {
+        assert(Thread.holdsLock(this))
+      }
+    }
+    nested
+  }
+  val synchInVal: Unit = synchronized {
+    assert(Thread.holdsLock(this))
+  }
+  lazy val synchInLazyVal: Unit = synchronized {
+    assert(Thread.holdsLock(this))
+  }
+  def method() = synchronized {
+    assert(Thread.holdsLock(this))
+  }
+}
+
+object Test {
+  // All of these were working, just double checking after the absense of synchronized in t9814.scala
+  def main(args: Array[String]): Unit = {
```

Comment:
These tests have nothing to do with specialization, right? Maybe a different test file name would be better. People reading the bug report would come back more confused